src/ex7/client/src/components/todos-container.js

```diff
@@ -1,34 +1,45 @@
-import React, { useState, useEffect, useCallback } from "react";
+import React, { useEffect, useCallback } from "react";
 import { ThreeDots } from  'react-loader-spinner'
-import { AddTodoForm } from "./add-todo-form.js";
-import { TodosList } from "./todos-list.js";
 import { getItems } from "../item-client.js";
+import AddTodoFormConnector from "../connectors/add-todo-form-connector";
+import TodosListConnector from "../connectors/todos-list-connector";
 import "../styles/todos-container.css";
 
-export function TodosContainer() {
-  const [todos, setTodos] = useState([]);
-  const [loaded, setLoaded] = useState(false)
+export function TodosContainer({loaded,
+                                serverErrorMessage, setServerErrorMessageAction,
+                                setTodosAction,
+                                setLoadedAction }) {
 
   const updateTodos = useCallback(async () => {
-    setTodos(await getItems());
-    setLoaded(true);
-  },[])
+    setServerErrorMessageAction("");
+    try {
+      setTodosAction(await getItems());
```

Comment:
use redux-thunk and move the reuqest inside setTodosAction

Reply:
moved inside